doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -877,30 +900,30 @@ if it reduces the subterm under consideration. The ``fold`` strategy takes
 a :token:`term` and tries to *unify* it to the current subterm, converting it to :token:`term`
 on success. It is stronger than the tactic ``fold``.
 
+.. _rewrite_strat_innermost_outermost:
 
-Usage
-~~~~~
+.. example:: :n:`innermost` and :n:`outermost`
 
+   The type of `Nat.add_comm` is `forall n m : nat, n + m = m + n`.
 
-.. tacn:: rewrite_strat @rewstrategy {? in @ident }
-   :name: rewrite_strat
+   .. coqtop:: all
 
-   Rewrite using the strategy s in hypothesis ident or the conclusion.
+      Require Import Coq.Arith.Arith.
+      Set Printing Parentheses.
+      Goal forall a b c: nat, a + b + c = 0.
+      rewrite_strat innermost <- Nat.add_comm.
 
-   .. exn:: Nothing to rewrite.
+   .. coqtop:: none
 
-      If the strategy failed.
-
-   .. exn:: No progress made.
-
-      If the strategy succeeded but made no progress.
+      Abort.
+      Goal forall a b c: nat, a + b + c = 0.
 
-   .. exn:: Unable to satisfy the rewriting constraints.
+   Using :n:`outermost` instead gives this result:
 
-      If the strategy succeeded and made progress but the
-      corresponding rewriting constraints are not satisfied.
+   .. coqtop:: all
 
+      rewrite_strat outermost <- Nat.add_comm.
```

Comment:
```suggestion
        rewrite_strat outermost Nat.add_comm.
```

Reply:
I couldn't see a difference in behavior with or without the `<-`, trying with:

```
Require Import Coq.Arith.Arith.
Set Printing Parentheses.
Goal forall a b c: nat, (a + b + c)*(a + b) = 0.
  rewrite_strat innermost Nat.add_comm.
```

If you can explain and it's simple, I'll add that, too.